R/pkg/R/DataFrame.R

```diff
@@ -246,30 +248,38 @@ setMethod("showDF",
 #' @note show(SparkDataFrame) since 1.4.0
 setMethod("show", "SparkDataFrame",
           function(object) {
-            allConf <- sparkR.conf()
-            if (!is.null(allConf[["spark.sql.repl.eagerEval.enabled"]]) &&
-                identical(allConf[["spark.sql.repl.eagerEval.enabled"]], "true")) {
-              argsList <- list()
-              argsList$x <- object
-              if (!is.null(allConf[["spark.sql.repl.eagerEval.maxNumRows"]])) {
-                numRows <- as.numeric(allConf[["spark.sql.repl.eagerEval.maxNumRows"]])
-                if (numRows > 0) {
-                  argsList$numRows <- numRows
+            showFunc <- getOption("sparkr.SparkDataFrame.base_show_func")
```

Comment:
hmm, this naming convention? typically we don't mix `.` and `_` and I don't think we have anything with `SparkDataFrame`

seems like we have `spark.sparkr.something` before https://github.com/apache/spark/blob/5264164a67df498b73facae207eda12ee133be7d/core/src/main/scala/org/apache/spark/api/r/RRunner.scala#L343

Reply:
I don't have any naming preference. One thing to note is that the `spark.sparkr.r.command` you mentioned above is the config to the Spark while the one I tried to define here is a config in the R side.

Reply:
IMO pretty print should plug in to something more R standard like 
[printr](https://yihui.name/printr/) or
[lemon](https://cran.r-project.org/web/packages/lemon/vignettes/lemon_print.html)
or
[print.x](https://stat.ethz.ch/R-manual/R-devel/library/base/html/print.dataframe.html)

Reply:
could we consider leaving print/show option out? I'd like to get eager compute to work even in basic sparkR / R shell

Reply:
It seems a good idea to separate out the print/show issue as I can see it is not 100% related to this jira. Maybe it will help with a new jira to track the requirement and discussion for that issue. Please confirm so that I will revert some related changes. Thanks.

Reply:
I have backed out the change of display function option. Also opened jira https://issues.apache.org/jira/browse/SPARK-25770 to track that issue.